syntax = "proto3";

package model;

option java_package = "com.chefencasa.model";
option java_outer_classname = "AccountProto";

import "google/protobuf/empty.proto";
import "google/protobuf/wrappers.proto";

service AccountsService {
  rpc FindByNumber(Account) returns (Account) {}
  rpc FindByCustomer(Account) returns (Accounts) {}
  rpc FindAll(google.protobuf.Empty) returns (Accounts) {}
  rpc AddAccount(Account) returns (Account) {}
}

message Accounts {
  repeated Account account = 1;
}

message Account {
  int32 id = 1;
  string number = 2;
  int32 customer_id = 3;
}